src/Bicep.Core/TypeSystem/ArmTemplateTypeLoader.cs

```diff
@@ -310,6 +311,12 @@ private static TypeSymbol GetObjectType(SchemaValidationContext context,
             {
                 var (type, typeName) = GetDeferrableTypeInfo(context, additionalPropertiesSchema);
                 additionalPropertiesType = type;
+                additionalPropertiesDescription = additionalPropertiesSchema.Metadata?.Value is JObject metadataObject &&
+                    metadataObject.TryGetValue(LanguageConstants.MetadataDescriptionPropertyName, out var descriptionToken) &&
+                    descriptionToken is JValue { Value: string descriptionString }
+                        ? descriptionString
+                        : null;
+
```

Comment:
Can this be factored into a function? The code to pull the description for each named property (l. 294-298) is very similar.